src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1434,7 +1435,7 @@ MasterExtendedOpNode(MultiExtendedOp *originalOpNode,
 		Expr *originalExpression = originalTargetEntry->expr;
 		Expr *newExpression = NULL;
 
-		bool hasAggregates = contain_agg_clause((Node *) originalExpression);
+		bool hasAggregates = contain_aggs_of_level((Node *) originalExpression, 0);
```

Comment:
These changes are probably somewhat obsolete after the recursive planning change. Reverting them does not yield test failures. 

I wonder whether it makes sense to keep the old call with the Assert since we really don't know how to handle Sublinks here, but either way works for me.

Reply:
Not included in this PR was not pushing down sublink arguments to aggregates. A test is included in this PR for `avg((subquery))`, but the expected result is `ERROR:  cannot handle unplanned sub-select`. This is being raised by `get_agg_clause_costs`. The rest of the fix will be deferring calls to `get_agg_clause_costs` until we process sublinks & not allowing sublinks to be pushed down. The sublinks after recursive planning are subqueries reading intermediate results, which obviously can't be pushed down

Reply:
Actually, intermediate results are typically broadcast to all nodes, so subqueries that read from them can be pushed down, assuming that does not create a correctness problem.